hsc/crawler.py

```diff
@@ -50,38 +89,34 @@ def get_submission_url(self, challenge_slug, submission_id):
 		return self.challenge_url.format(challenge_slug, submission_id)
 
 	def store_submission(self, file_name, code):
-		print(file_name)
 		os.makedirs(os.path.dirname(file_name), exist_ok=True)
 		with open(file_name, 'w') as text_file:
-			print(code, file=text_file)
-
-	def update_readme(self, challenge_name, readme_file_path, challenge_slug, file_name, file_extension):
-		problem_url = self.problem_url.format(challenge_slug)
-		text = self.problem_readme_text.format(challenge_name, problem_url, file_name, file_extension)
-		with open(readme_file_path, 'a') as text_file:
-			print(text, file=text_file)
-		with open(readme_file_path, 'r') as text_file:
+			text_file.write(code)
+
+	def update_readme(self, readme_file_path, problem_readme_text):
+		with open(readme_file_path, 'r+') as text_file:
 			lines = text_file.readlines()
+			lines.append(problem_readme_text)
 			sortedlines = lines[:4] + sorted(lines[4:])
-		with open(readme_file_path, 'w') as text_file:
+			text_file.seek(0)
 			text_file.writelines(sortedlines)
```

Comment:
Is there no requirement of opening the file in write mode first?

Reply:
r+ is read write mode